src/main/java/com/meilisearch/sdk/Config.java

```diff
@@ -1,20 +1,23 @@
 package com.meilisearch.sdk;
 
+import com.meilisearch.sdk.json.GsonJsonHandler;
+import com.meilisearch.sdk.json.JsonHandler;
 import lombok.Getter;
 
 /** Meilisearch configuration */
 @Getter
 public class Config {
-    String hostUrl;
-    String apiKey;
+    protected final String hostUrl;
+    protected final String apiKey;
+    protected final JsonHandler jsonHandler;
 
     /**
      * Creates a configuration without an API key
      *
      * @param hostUrl URL of the Meilisearch instance
      */
     public Config(String hostUrl) {
-        this(hostUrl, "");
+        this(hostUrl, "", new GsonJsonHandler());
```

Comment:
> I'm not sure about this constructor. It assumes the apiKey is empty, sounds a little bit reckless.

It does, but `Meilisearch` supports the ability to run without any key, so the SDK can do that too.

> About the idea of calling Gson directly

Yes ! My idea was to have `Gson` by default so we have this one hard written and we advise to use this one but if the user want to use another one he could change it. The problem is that if we don't write a hardcoded choice, the user will have to instantiate one when creating a Client and this will add a step, which is not very recommendable for most users and would add complexity.